mira/modeling/ode.py

```diff
@@ -131,9 +132,24 @@ def simulate_ode_model(ode_model: OdeModel, times, initials=None,
     solver = scipy.integrate.ode(f=rhs)
 
     # initials is a list
-    solver.set_initial_value(initials)
+    initial_values = []
+
+    # If there are type of sympy.Integer or sympy.Float, convert to regular int or float
+    for expression in initials:
+        if isinstance(expression, SympyExprStr):
+            if isinstance(expression.args[0], sympy.Integer):
+                initial_values.append(int(expression.args[0]))
+            elif isinstance(expression, sympy.Float):
+                initial_values.append(float(expression.args[0]))
+        elif isinstance(expression,sympy.Float):
+            initial_values.append(float(expression.args[0]))
+        elif isinstance(expression,sympy.Integer):
+            initial_values.append(int(expression.args[0]))
+        else:
+            initial_values.append(expression)
```

Comment:
I don't think this part above is right. First, we need to _evaluate_ the expression by substituting parameter values to get a specific number. Then we just use that number as the initial value. No need to keep track of ints, floats, etc.

Reply:
```
    for expression in initials:
        if isinstance(expression,sympy.Expr):
            if isinstance(expression.args[0], sympy.Float):
                expression = float(expression.subs(expression, expression.args[0]))
            elif isinstance(expression.args[0], sympy.Integer):
                expression = int(expression.subs(expression, expression.args[0]))
        initial_values.append(expression)
```
You mentioned how we don't need to keep track of floats and ints, but using the `subs` method returns a SympyExprStr object rather than a primitive float or int, so to make sure that the typing is preserved when we call `simulate_ode_model`, I test to see if the expression is of type `sympy.Float` or `sympy.Integer`.